### PR TITLE
Switch more tests to LF 1.16

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

File: sdk/compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml [hunk 1]
```diff
@@ -110,10 +110,6 @@ type Choice t c r =
   , HasExercise t c r
   , HasToAnyChoice t c r
   , HasFromAnyChoice t c r
-#ifdef DAML_CHOICE_FUNCS
-  , HasChoiceController t c
-  , HasChoiceObserver t c
-#endif -- DAML_CHOICE_FUNCS
```
Comment (remyhaemmerle-da) [inline sdk/compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml:116]: Hey @moisesackerman-da why do we need those in the `Choice` alias.
With this I cannot plug LF 1.dev with 1.16.

Comment (moisesackerman-da) [reply sdk/compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml:116]: they're not strictly needed, I think the idea was just that users could use them with a `Choice t c r` constraint instead of the longer `HasChoiceController t c`/`HasChoiceObserver t c`.

> With this I cannot plug LF 1.dev with 1.16

I think that means that some instances of LF 1.dev have to stay as 1.dev instead of being replaced with 1.16, since they need the `DAML_CHOICE_FUNCS` feature. What breaks if you keep these lines?

Comment (remyhaemmerle-da) [reply sdk/compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml:116]: for instance the daml script
```
internalTypedExerciseCmd : forall t c r. Choice t c r => ContractId t -> c -> Bool -> Commands r
``` 
compiled in 1.dev will require instances for  `HasChoiceController t c`  `HasChoiceObserver t c`, but they are not generated for <= 1.16.

File: sdk/daml-script/test/BUILD.bazel [hunk 2]
```diff
@@ -61,18 +63,16 @@ dependencies:
...
-            major = major,
             sdk = sdk_version,
             target = target,
         ),
         tools = ["//compiler/damlc"],
         visibility = ["//visibility:public"],
     )
-    for major in LF_MAJOR_VERSIONS
-    for target in [lf_version_default_or_latest(major)]
+    for target in [lf_version_configuration.get("latest")]
```
Comment (samuel-williams-da) [inline sdk/daml-script/test/BUILD.bazel:75]: This doesn't really need to be a loop anymore, are you keeping it with intend to add more cases later?

File: sdk/daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/MultiUpgradesIT.scala [hunk 3]
```diff
@@ -13,16 +13,14 @@ import org.scalatest.Inside
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AsyncWordSpec
 
-class UpgradesITDev extends AsyncWordSpec with AbstractScriptTest with Inside with Matchers {
+class MultiUpgradesIT extends AsyncWordSpec with AbstractScriptTest with Inside with Matchers {
```
Comment (samuel-williams-da) [inline sdk/daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/MultiUpgradesIT.scala:16]: Note for future: we'll update this to match `main`, rename to `RuntimeUpgradesIT`, and perhaps even remove the existing `UpgradesIT`

File: sdk/ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_16.scala [hunk 4]
```diff
@@ -8,11 +8,8 @@ import com.daml.ledger.api.tls.TlsConfiguration
 
 package object v1_16 {
   def default(timeoutScaleFactor: Double): Vector[LedgerTestSuite] =
-    v1_15.default(timeoutScaleFactor)
+    v1_15.default(timeoutScaleFactor) ++ Vector(new UpgradingIT)
```
Comment (samuel-williams-da) [inline sdk/ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_16.scala:11]: Have you verified that this doesn't fail? The comment before implied it did

File: sdk/daml-lf/data/src/main/scala/com/daml/data/FixedOrder.scala
Comment (samuel-williams-da) [inline sdk/daml-lf/data/src/main/scala/com/daml/data/FixedOrder.scala:1]: I'm a little confused where this file came from, and where its used.

File: sdk/compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml (hunk 1 above)
Comment (samuel-williams-da) [reply sdk/compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml:116]: I don't totally follow whats happening here, if the constraints in daml3-script need changing though, thats fine?

**Review by samuel-williams-da — APPROVED** [review]

File: sdk/compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml (hunk 1 above)
Comment (moisesackerman-da) [reply sdk/compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml:116]: Ah, I see it now. If package `A` is compiled with LF 1.16, and I import it as a data-dependency from a package `B` compiled with LF 1.dev, I won't be able to call e.g. `exerciseCmd : forall t c r. Choice t c r => ContractId t -> c -> Commands r` on a template from `A`, because `Choice t c r` now includes a constraint corresponding to an instance that was never generated for `A` templates, and data-dependencies does not add that instance (and I don't think it could). 

tl;dr: let's remove these from `type Choice`, thank you Rémy

File: sdk/daml-script/test/BUILD.bazel (hunk 2 above)
Comment (remyhaemmerle-da) [reply sdk/daml-script/test/BUILD.bazel:75]: never really needed a loop.
We use the loop comprehension syntax to "emulate" local variable (something like `where target = ...`  in Haskell).

File: sdk/ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_16.scala (hunk 4 above)
Comment (remyhaemmerle-da) [reply sdk/ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_16.scala:11]: Suggested by @tudor-da, so it should be fine.

File: sdk/daml-lf/data/src/main/scala/com/daml/data/FixedOrder.scala
Comment (remyhaemmerle-da) [reply sdk/daml-lf/data/src/main/scala/com/daml/data/FixedOrder.scala:1]: :grimacing: 
I delete that immediately.
